### PR TITLE
[plugin.video.dailymotion_com] 2.3.9

### PR DESCRIPTION
**Pull request by Gujal00**
### Description
 - Make code Python 3 compatible
 - New artwork by jokstergfx

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Gujal00** [comment]
Please hold review, the Live Channel playback seem to be broken today. Will debug and let you know when working

File: plugin.video.dailymotion_com/default.py [hunk 1]
```diff
@@ -62,42 +67,50 @@ def strip_tags(html):
...
+    addDir(translation(30002) + " " + translation(30007), "", 'usersearch', _ipath + "search_users.png")
     if dmUser:
-        addDir(translation(30034), "", "personalMain", "")
+        addDir(translation(30034), "", "personalMain", _ipath + "my_stuff.png")
     else:
-        addFavDir(translation(30024), "", "favouriteUsers", "")
+        addFavDir(translation(30024), "", "favouriteUsers", _ipath + "favourite_users.png")
+    xbmcplugin.setContent(pluginhandle, "addons")
     xbmcplugin.endOfDirectory(pluginhandle)
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```
Comment (Rechi) [inline plugin.video.dailymotion_com/default.py]: This isn't allowed.

File: plugin.video.dailymotion_com/default.py [hunk 2]
```diff
@@ -62,42 +67,50 @@ def strip_tags(html):
...
+    addDir(translation(30036), urlMain+"/user/"+dmUser+"/subscriptions?fields=description,duration,id,owner.username,taken_time,thumbnail_large_url,title,views_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listVideos', _ipath + "following.png")
+    addDir(translation(30037), urlMain+"/user/"+dmUser+"/favorites?fields=description,duration,id,owner.username,taken_time,thumbnail_large_url,title,views_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listVideos', _ipath + "favourites.png")
+    addDir(translation(30038), urlMain+"/user/"+dmUser+"/playlists?fields=id,name,videos_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listUserPlaylists', _ipath + "playlists.png")   
+    xbmcplugin.setContent(pluginhandle, 'addons')
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```
Comment (Rechi) [inline plugin.video.dailymotion_com/default.py]: This isn't allowed.

File: plugin.video.dailymotion_com/default.py [hunk 3]
```diff
@@ -113,16 +126,18 @@ def favouriteUsers():
           match = re.compile('###USER###=(.+?)###THUMB###=(.*?)###END###', re.DOTALL).findall(content)
           for user, thumb in match:
             addUserFavDir(user, 'owner:'+user, 'sortVideos1', thumb)
+    xbmcplugin.setContent(pluginhandle, "addons")
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```
Comment (Rechi) [inline plugin.video.dailymotion_com/default.py]: This isn't allowed.

File: plugin.video.dailymotion_com/default.py [hunk 4]
```diff
@@ -132,19 +147,23 @@ def sortVideos1(url):
...
-    addDir(translation(30008), url, 'listVideos', "")
-    addDir(translation(30009), url.replace("sort=recent", "sort=visited"), 'sortVideos2', "")
+    addDir(translation(30015), url.replace("sort=recent", "sort=trending"), 'listVideos', _ipath + "trending.png")
+    addDir(translation(30008), url, 'listVideos', _ipath + "most_recent.png")
+    addDir(translation(30009), url.replace("sort=recent", "sort=visited"), 'sortVideos2', _ipath + "most_viewed.png")
     if type == "owner":
-        addDir("- "+translation(30038), urlMain+"/user/"+gid+"/playlists?fields=id,name,videos_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listUserPlaylists', "")
+        addDir("- "+translation(30038), urlMain+"/user/"+gid+"/playlists?fields=id,name,videos_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listUserPlaylists', _ipath + "playlists.png")
+    xbmcplugin.setContent(pluginhandle, 'addons')
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```
Comment (Rechi) [inline plugin.video.dailymotion_com/default.py]: This isn't allowed.

File: plugin.video.dailymotion_com/default.py [hunk 5]
```diff
@@ -132,19 +147,23 @@ def sortVideos1(url):
...
-    addDir(translation(30012), url.replace("sort=visited", "sort=visited-week"), "listVideos", "")
-    addDir(translation(30013), url.replace("sort=visited", "sort=visited-month"), "listVideos", "")
-    addDir(translation(30014), url, 'listVideos', "")
+    addDir(translation(30010), url.replace("sort=visited", "sort=visited-hour"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30011), url.replace("sort=visited", "sort=visited-today"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30012), url.replace("sort=visited", "sort=visited-week"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30013), url.replace("sort=visited", "sort=visited-month"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30014), url, 'listVideos', _ipath + "most_viewed.png")
+    xbmcplugin.setContent(pluginhandle, 'addons')
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```
Comment (Rechi) [inline plugin.video.dailymotion_com/default.py]: This isn't allowed.

File: plugin.video.dailymotion_com/default.py [hunk 6]
```diff
@@ -197,45 +217,45 @@ def listVideos(url):
     if content['has_more']:
         currentPage = content['page']
         nextPage = currentPage+1
-        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listVideos', "")
-    if forceViewModeNew:
-        xbmc.executebuiltin('Container.SetViewMode('+viewModeNew+')')
+        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listVideos', _ipath + "next_page2.png")
+    xbmcplugin.setContent(pluginhandle, "episodes")
+    xbmc.executebuiltin('Container.SetViewMode(54)')
```
Comment (Rechi) [inline plugin.video.dailymotion_com/default.py]: This isn't allowed.

File: plugin.video.dailymotion_com/default.py [hunk 7]
```diff
@@ -197,45 +217,45 @@ def listVideos(url):
...
             videos = item['videos_total']
             views = item['views_total']
             addUserDir(user, 'owner:'+user, 'sortVideos1', thumb.replace("\\", ""), "Views: "+str(views)+"\nVideos: "+str(videos))
     if content['has_more']:
         currentPage = content['page']
         nextPage = currentPage+1
-        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listUsers', "")
+        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listUsers', _ipath + "next_page.png")
     xbmcplugin.setContent(pluginhandle, "addons")
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```
Comment (Rechi) [inline plugin.video.dailymotion_com/default.py]: This isn't allowed.

File: plugin.video.dailymotion_com/default.py [hunk 8]
```diff
@@ -197,45 +217,45 @@ def listVideos(url):
...
         thumb = item['thumbnail_large_url']
         views = item['views_last_hour']
         addLiveLink(title, vid, 'playLiveVideo', thumb.replace("\\", ""), views)
     if content['has_more']:
         currentPage = content['page']
         nextPage = currentPage+1
-        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listLive', "")
+        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listLive', _ipath + "next_page2.png")
     xbmcplugin.setContent(pluginhandle, "episodes")
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```
Comment (Rechi) [inline plugin.video.dailymotion_com/default.py]: This isn't allowed.

File: plugin.video.dailymotion_com/default.py (hunk 6 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Can you clarify, please
The addon had this this line before
`xbmc.executebuiltin('Container.SetViewMode('+viewModeNew+')')`
and I replaced it with 
`xbmc.executebuiltin('Container.SetViewMode(54)')`

Is SetViewMode not allowed anymore? it does work correctly as intended though.

File: plugin.video.dailymotion_com/default.py (hunk 1 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Corrected it to use only if user setup the relevant modes in settings

File: plugin.video.dailymotion_com/default.py (hunk 2 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Corrected it to use only if user setup the relevant modes in settings

File: plugin.video.dailymotion_com/default.py (hunk 3 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Corrected it to use only if user setup the relevant modes in settings

File: plugin.video.dailymotion_com/default.py (hunk 4 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Corrected it to use only if user setup the relevant modes in settings

File: plugin.video.dailymotion_com/default.py (hunk 5 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Corrected it to use only if user setup the relevant modes in settings

File: plugin.video.dailymotion_com/default.py (hunk 6 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Corrected it to use only if user setup the relevant modes in settings

File: plugin.video.dailymotion_com/default.py (hunk 7 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Corrected it to use only if user setup the relevant modes in settings

File: plugin.video.dailymotion_com/default.py (hunk 8 above)
Comment (Gujal00) [reply plugin.video.dailymotion_com/default.py]: Corrected it to use only if user setup the relevant modes in settings

File: plugin.video.dailymotion_com/resources/settings.xml [hunk 9]
```diff
@@ -9,9 +9,14 @@
 	<category label="30112">
 		<setting id="downloadDir" type="folder" label="30108" default=""/>
 	</category>
+	<category label="30102">
+		<setting id="forceViewMode" type="bool" label="30102" default="true"/>
```
Comment (Rechi) [inline plugin.video.dailymotion_com/resources/settings.xml]: It's okay to allow add a configure setting for forcing the view mode, but it has to be off by default.

Comment (Gujal00) [reply plugin.video.dailymotion_com/resources/settings.xml]: Thanks for the clarification. Changed and squashed the commit.

**Pull request merged**